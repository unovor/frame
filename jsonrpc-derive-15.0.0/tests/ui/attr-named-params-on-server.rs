use jsonrpc_derive::rpc;

#[rpc]
pub trait Rpc {
	/// Returns a protocol version
	#[rpc(name = "add", params = "named")]
	fn add(&self, a: u32, b: u32) -> Result<String>;
}

fn main() {}
