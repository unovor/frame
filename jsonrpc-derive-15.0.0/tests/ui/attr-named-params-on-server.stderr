error: `params = "named"` can only be used to generate a client (on a trait annotated with #[rpc(client)]). At this time the server does not support named parameters.
 --> $DIR/attr-named-params-on-server.rs:4:1
  |
4 | / pub trait Rpc {
5 | |     /// Returns a protocol version
6 | |     #[rpc(name = "add", params = "named")]
7 | |     fn add(&self, a: u32, b: u32) -> Result<String>;
8 | | }
  | |_^
